### PR TITLE
feature: update Iroha image version, recompiled executor and nft smart contract, added test for trigger metadata update event

### PR DESCRIPTION
**Pull request by gv-timur**
update Iroha image version, recompiled executor and nft smart contract, added test for trigger metadata update event

### CODE REVIEW

**Review by Mingela — DISMISSED** [review]

File: modules/client/src/test/kotlin/jp/co/soramitsu/iroha2/TriggersTest.kt [hunk 1]
```diff
@@ -253,44 +261,89 @@ class TriggersTest : IrohaTest<Iroha2Client>() {
...
+
+    @Test
+    @WithIroha([AliceHas100XorAndPermissionToBurn::class])
+    @Story("Wasm trigger mints NFT for every user when trigger metadata is updated")
+    @SdkTestId("wasm_trigger_to_mint_nft_for_every_user_on_update_trigger_metadata_event")
+    fun `wasm trigger to mint nft for every user on update trigger metadata event`(): Unit = runBlocking {
+        val wasmTriggerId = TriggerId(name = "wasm_trigger".asName())
+        val setKeyValueTriggerId = TriggerId(name = "update_trigger".asName())
+
+        val filter = TriggeringFilterBox.Data(
```
Comment (arndey) [inline modules/client/src/test/kotlin/jp/co/soramitsu/iroha2/TriggersTest.kt:290]: please use our object Filters

File: modules/client/src/test/kotlin/jp/co/soramitsu/iroha2/TriggersTest.kt [hunk 2]
```diff
@@ -253,44 +261,89 @@ class TriggersTest : IrohaTest<Iroha2Client>() {
...
-                setKeyValue(ALICE_ACCOUNT_ID, "test$i".asName(), "test$i".asValue())
-                buildSigned(ALICE_KEYPAIR)
-            }.also { d ->
-                withTimeout(txTimeout) { d.await() }
-            }
+        keepNetworkBusyAndCheckAssetDefinitionIds()
+
+        val testKey = "key"
+        val testValue = "value"
+        client.sendTransaction {
```
Comment (arndey) [inline modules/client/src/test/kotlin/jp/co/soramitsu/iroha2/TriggersTest.kt:268]: client.tx

File: modules/client/src/test/kotlin/jp/co/soramitsu/iroha2/TriggersTest.kt (hunk 1 above)
Comment (gv-timur) [reply modules/client/src/test/kotlin/jp/co/soramitsu/iroha2/TriggersTest.kt]: fixed

File: modules/client/src/test/kotlin/jp/co/soramitsu/iroha2/TriggersTest.kt (hunk 2 above)
Comment (gv-timur) [reply modules/client/src/test/kotlin/jp/co/soramitsu/iroha2/TriggersTest.kt]: fixed

**Review by Mingela — APPROVED** [review]

File: modules/client/src/test/kotlin/jp/co/soramitsu/iroha2/TriggersTest.kt [hunk 3]
```diff
@@ -248,49 +254,75 @@ class TriggersTest : IrohaTest<Iroha2Client>() {
...
+    @Test
+    @WithIroha([AliceHas100XorAndPermissionToBurn::class])
+    @Story("Wasm trigger mints NFT for every user when trigger metadata is updated")
+    @SdkTestId("wasm_trigger_to_mint_nft_for_every_user_on_update_trigger_metadata_event")
+    fun `wasm trigger to mint nft for every user on update trigger metadata event`(): Unit = runBlocking {
+        val wasmTriggerId = TriggerId(name = "wasm_trigger".asName())
+        val setKeyValueTriggerId = TriggerId(name = "update_trigger".asName())
+
+        val filter = Filters.data(
+            DataEntityFilter.ByTrigger(
```
Comment (arndey) [inline modules/client/src/test/kotlin/jp/co/soramitsu/iroha2/TriggersTest.kt]: The point is not to replace one line, but to make the expression smaller overall. It should look like `Filters.byTrigger(wasmTriggerId, ...)`

Comment (gv-timur) [reply modules/client/src/test/kotlin/jp/co/soramitsu/iroha2/TriggersTest.kt]: fixed

**Review by arndey — APPROVED** [review]

**Pull request merged**